src/common/vscode/tree/TreeItem.ts

```diff
@@ -23,4 +27,18 @@ export default class TreeItem extends vscode.TreeItem {
     public getParent(): TreeItem | undefined {
         return this.parent;
     }
+
+    protected async setRemoteIconPath(path: string | undefined) {
+        if (path) {
+            const cachedCroppedImageUrl = getCachedCroppedImageUrl(path);
+            if (cachedCroppedImageUrl) {
+                this.iconPath = vscode.Uri.parse(cachedCroppedImageUrl);
+            }
```

Comment:
So, shouldn't we need to return here to stop making the request below?

Reply:
I had something else in my mind but changed it to the way you described